File: get_logos.py
import urllib.request
import json

files = {
    "Havells": "File:Havells_Logo.svg",
    "A. O. Smith": "File:A.O._Smith_logo.svg",
    "Pentair": "File:Pentair_logo.svg",
    "Adani": "File:Adani_logo.svg",
    "Tata": "File:Tata_logo.svg",
    "Crompton": "File:Crompton_Greaves_Logo.svg",
    "Grundfos": "File:Grundfos_Logo.svg",
}

urls = {}
for brand, filename in files.items():
    try:
        url = f"https://en.wikipedia.org/w/api.php?action=query&titles={urllib.parse.quote(filename)}&prop=imageinfo&iiprop=url&format=json"
        req = urllib.request.urlopen(url)
        res = json.loads(req.read())
        pages = res['query']['pages']
        page = list(pages.values())[0]
        image_url = page['imageinfo'][0]['url']
        urls[brand] = image_url
    except Exception as e:
        urls[brand] = str(e)

print(json.dumps(urls, indent=2))

File: get_logos.js
const fs = require('fs');
const https = require('https');

const files = {
    "Havells_Logo.svg": "https://upload.wikimedia.org/wikipedia/commons/4/4b/Havells_Logo.svg",
    "A.O._Smith_logo.svg": "https://upload.wikimedia.org/wikipedia/commons/e/eb/A.O._Smith_logo.svg",
    "Pentair_logo.svg": "https://upload.wikimedia.org/wikipedia/commons/e/e9/Pentair_logo.svg",
    "Adani_logo.svg": "https://upload.wikimedia.org/wikipedia/commons/0/07/Adani_logo.svg",
    "Tata_logo.svg": "https://upload.wikimedia.org/wikipedia/commons/b/b1/Tata_logo.svg",
    "Crompton_Greaves_Logo.svg": "https://upload.wikimedia.org/wikipedia/commons/5/52/Crompton_Greaves_Logo.svg",
    "Grundfos_Logo.svg": "https://upload.wikimedia.org/wikipedia/commons/6/6c/Grundfos_Logo.svg"
};

const options = {
    headers: {
        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36'
    }
};

for (const [filename, url] of Object.entries(files)) {
    https.get(url, options, (res) => {
        let data = '';
        res.on('data', chunk => data += chunk);
        res.on('end', () => {
            fs.writeFileSync('brands/' + filename, data);
            console.log('Downloaded: ' + filename + ' (' + data.length + ' bytes)');
        });
    });
}
